src/common.js

```diff
@@ -274,7 +283,24 @@ function wrap(cmd, fn, options) {
         });
         if (!config.noglob && options && typeof options.idx === 'number')
           args = args.slice(0, options.idx).concat(expand(args.slice(options.idx)));
-        retValue = fn.apply(this, args);
+        if (config.async || callback) {
+          process.nextTick(function() {
+            var r = fn.apply(this, args);
```

Comment:
Why would this not work? The function won't be called until everything on the event loop is dealt with, right?